### PR TITLE
Bluetooth: host: Added inline function for gatt notify by uuid

### PR DESCRIPTION
**Pull request by Emil-Gydesen-Bose**
Added a static inline helper function to notify by uuid, similar to `bt_gatt_notify`.

### CODE REVIEW

**Review by Vudentz — APPROVED** [review]
LGTM

**Comment by Emil-Gydesen-Bose** [comment]
I fixed the warnings from checkpath, but re-pushing apparently didn't retrigger it. Can I manually request a re-run somehow?

**Comment by joerchan** [comment]
@Emil-Gydesen-Bose No I think checkpatch is still right. There is tabs in this part:
```
 *  @param uuid	The UUID. If the server contains multiple services with the same
 * 		UUID, then the first occurrence, starting from the attr given,
 * 		is used.
 *  @param attr	Pointer to an attribute that serves as the starting point for
 * 		the search of a match for the UUID.
```
You should align using spaces only here.

File: include/bluetooth/gatt.h [hunk 1]
```diff
@@ -917,6 +917,43 @@ static inline int bt_gatt_notify(struct bt_conn *conn,
...
+ *  of the UUID.
+ *
+ *  @param conn	Connection object.
+ *  @param uuid	The UUID. If the server contains multiple services with the same
+ * 		UUID, then the first occurrence, starting from the attr given,
+ * 		is used.
+ *  @param attr	Pointer to an attribute that serves as the starting point for
+ * 		the search of a match for the UUID.
+ *  @param data	Pointer to Attribute data.
+ *  @param len	Attribute value length.
```
Comment (joerchan) [inline include/bluetooth/gatt.h:936]: All the tabs in this region should be replaced with spaces.

File: include/bluetooth/gatt.h [hunk 2]
```diff
@@ -917,6 +917,43 @@ static inline int bt_gatt_notify(struct bt_conn *conn,
 	return bt_gatt_notify_cb(conn, &params);
 }
 
+/** @brief Notify attribute value change by UUID.
+ *
+ *  Send notification of attribute value change, if connection is NULL notify
+ *  all peer that have notification enabled via CCC otherwise do a direct
+ *  notification only on the given connection.
+ *
+ *  The attribute object on the parameters is the starting point for the search
```
Comment (joerchan) [inline include/bluetooth/gatt.h:926]: i think we should remove "on the parameters", the attribute is not part of a params struct.

**Review by joerchan — CHANGES_REQUESTED** [review]

File: include/bluetooth/gatt.h (hunk 2 above)
Comment (Emil-Gydesen-Bose) [reply include/bluetooth/gatt.h:926]: Agreed

File: include/bluetooth/gatt.h (hunk 1 above)
Comment (Emil-Gydesen-Bose) [reply include/bluetooth/gatt.h:936]: Fixed

Comment (Emil-Gydesen-Bose) [reply include/bluetooth/gatt.h]: Ah, yeah. Surprised that wasn't caught by the CI.

Comment (Emil-Gydesen-Bose) [reply include/bluetooth/gatt.h]: Should be fixed now :) Thanks for catching that

Comment (joerchan) [reply include/bluetooth/gatt.h]: CI doesn't catch it unfortunately.

**Review by joerchan — APPROVED** [review]

**Pull request merged**